analytics-listeners/src/main/java/org/exoplatform/analytics/listener/tasks/TaskLabelListener.java

```diff
@@ -0,0 +1,73 @@
+package org.exoplatform.analytics.listener.tasks;
+
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.analytics.model.StatisticData;
+import org.exoplatform.analytics.utils.AnalyticsUtils;
+import org.exoplatform.container.PortalContainer;
+import org.exoplatform.container.component.RequestLifeCycle;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.listener.Asynchronous;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.task.domain.LabelTaskMapping;
+import org.exoplatform.task.service.LabelService;
+import org.stringtemplate.v4.ST;
+
+@Asynchronous
+public class TaskLabelListener extends Listener<String, LabelTaskMapping> {
+
+  private PortalContainer container;
+
+  private IdentityManager identityManager;
+
+  private final String    operation;
+
+  public TaskLabelListener(InitParams params) {
+    this.container = PortalContainer.getInstance();
+    this.operation = params.getValueParam("operation").getValue();
+  }
+
+  @Override
+  public void onEvent(Event<String, LabelTaskMapping> event) throws Exception {
+    RequestLifeCycle.begin(container);
+    LabelTaskMapping labelTaskMapping = event.getData();
+    String userName = event.getSource();
+    createLabelStatistic(labelTaskMapping, userName);
+    RequestLifeCycle.end();
```

Comment:
Open transaction before try block + add the end transaction to close it (even when there is an error or rollback) inside a finally + add ExoContainerContext.setCurrent to make sure that JPA transactions are correctly started.
```suggestion
    RequestLifeCycle.begin(container);
    ExoContainerContext.setCurrent(container);
    try {
      LabelTaskMapping labelTaskMapping = event.getData();
      String userName = event.getSource();
      createLabelStatistic(labelTaskMapping, userName);
    } finally {
      RequestLifeCycle.end();
    }
```

Reply:
you can as well, just add `@ExoTransactional` annotation to this **public** method + ensure that you have the jcabi maven plugin enabled in the pom.xml to ensure that the annotation is processed in compilation time.